tests/pytorch/r1.13/common.libsonnet

```diff
@@ -24,45 +24,118 @@ local volumes = import 'templates/volumes.libsonnet';
+            env+: [
+              {
+                name: 'TPU_NAME',
+                valueFrom: {
+                  fieldRef: {
+                    fieldPath: "metadata.annotations['name.cloud-tpus.google.com/train']",
+                  },
+                },
+              },
+            ],
+            command: [
+              'python3',
+              '-c',
+              |||
+                import importlib_metadata
+                import os
+                import re
+
+                import cloud_tpu_client
+
+                requirements = importlib_metadata.requires('torch_xla')
+                libtpu_pattern = r'libtpu-nightly ?@ https:\/\/storage.googleapis.com\/cloud-tpu-tpuvm-artifacts\/wheels\/libtpu-nightly\/libtpu_nightly-\d.\d.dev(\d{8})-\w+-\w+-\w+.whl'
+                libtpu_matches = [
+                  re.findall(libtpu_pattern, req)[0]
+                  for req in requirements
+                  if re.match(libtpu_pattern, req)
+                ]
+                assert len(libtpu_matches) == 1, f'{len(libtpu_matches)} matches in {requirements} (pattern: `{libtpu_pattern}`)'
+                libtpu_date = libtpu_matches[0]
+                print('libtpu date:', libtpu_date)
+
+                ctc = cloud_tpu_client.Client(tpu=os.path.basename('$(TPU_NAME)'), zone=os.path.dirname('$(TPU_NAME)'))
+                ctc.wait_for_healthy()
+                ctc.configure_tpu_version(f'pytorch-nightly-dev{libtpu_date}', restart_type='always')
+                ctc.wait_for_healthy()
+              |||,
+            ],
+          },
+        },
+      },
+    },
   },
-  PyTorchTest:: common.PyTorchTest + r1_13,
   PyTorchXlaDistPodTest:: common.PyTorchXlaDistPodTest + r1_13,
   PyTorchGkePodTest:: common.PyTorchGkePodTest + r1_13,
   Functional:: mixins.Functional {
-    schedule: '0 7 * * 1',
+    schedule: '0 7 * * *',
     tpuSettings+: {
       preemptible: false,
     },
   },
-  Convergence:: mixins.Convergence {
-    schedule: null,
-  },
+  Convergence:: mixins.Convergence,
   PyTorchTpuVmMixin:: experimental.PyTorchTpuVmMixin {
+    local config = self,
+
     tpuSettings+: {
-      softwareVersion: 'tpu-vm-base',
+      softwareVersion: if config.accelerator.version < 4 then
+        'tpu-vm-base'
+      else
+        'tpu-vm-v4-base',
       tpuVmPytorchSetup: |||
-        sudo pip3 uninstall --yes torch torch_xla torchvision numpy
-        sudo pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch-1.13-cp38-cp38-linux_x86_64.whl
-        sudo pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-1.13-cp38-cp38-linux_x86_64.whl
-        sudo pip3 install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torchvision-1.13-cp38-cp38-linux_x86_64.whl
-        sudo pip3 install https://storage.googleapis.com/cloud-tpu-tpuvm-artifacts/wheels/libtpu-nightly/libtpu_nightly-0.1.dev20220930-py3-none-any.whl
-        sudo pip3 install numpy
```

Comment:
wondering if we will need these..

Reply:
I think they're replaced by https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/df347546f2ff0c71ac7acbc25babdf9ef985216d/tests/pytorch/nightly/common.libsonnet#L98-L100

Reply:
sorry I meant the
```
sudo pip3 install mkl mkl-include cloud-tpu-client
        sudo apt-get -y update
        sudo apt-get install -y libomp5
```

Reply:
I guess to validate, we'll have to run all tests? But if it fails, we'd lose one day? Do you want me to add them anyway?